src/app/core/configurable-enum/configurable-enum.module.ts

```diff
@@ -0,0 +1,63 @@
+import { NgModule } from "@angular/core";
+import { CommonModule } from "@angular/common";
+import { ConfigService } from "../config/config.service";
+import { EntitySchemaService } from "../entity/schema/entity-schema.service";
+import { ConfigurableEnumDatatype } from "./configurable-enum-datatype/configurable-enum-datatype";
+import { ConfigurableEnumDirective } from "./configurable-enum-directive/configurable-enum.directive";
+
+/**
+ * Provides a generic functionality to define enums (collections of selectable options) in the config database
+ * and save them efficiently as `@DatabaseField`s.
+ *
+ * An enum in the config can look something like this:
+```
+ "enum:interaction-type": [
+   {
+     "id": "",
+     "name": ""
+   },
+   {
+     "id": "HOME_VISIT",
+     "name": "Home Visit"
+   },
+   {
+     "id": "GUARDIAN_TALK",
+     "name": "Talk with Guardians"
+   },
+ ]
+```
+ *
+ * In an entity a property can then be of this type:
+ * `@DatabaseField({ dataType: "configurable-enum", innerDataType: INTERACTION_TYPE_CONFIG_ID }) category: InteractionType;`
+ *
+ * ConfigurableEnum values can include arbitrary additional properties but must at least have basic properties of `ConfigurableEnumValue` interface.
+ * It is best practice to define an interface for more complex value types and define a constant for the enum's config id, e.g.
+```
+ export interface InteractionType extends ConfigurableEnumValue {
+  id: string;
+  label: string;
+  color?: string;
+  isMeeting?: boolean;
+ }
+
+ export const INTERACTION_TYPE_CONFIG_ID = "interaction-type";
+```
+ *
+ * To iterate over all enum values in a template (e.g. to set up a mat-select dropdown) you can use
+ * {@link ConfigurableEnumDirective} similar to `*ngFor`.
+ */
+@NgModule({
+  declarations: [ConfigurableEnumDirective],
+  imports: [CommonModule],
+  exports: [ConfigurableEnumDirective],
+})
+export class ConfigurableEnumModule {
+  constructor(
+    private configService: ConfigService,
+    private entitySchemaService: EntitySchemaService
+  ) {
+    this.entitySchemaService.registerSchemaDatatype(
+      new ConfigurableEnumDatatype(configService)
```

Comment:
I think there is a problem. If a component does not use the `ConfigurableEnumDirective`, the constructor of this module is not executed. This means if the component loads an entity with an `ConfigurableEnumDatatype` attribute, the schema service will not be able to parse that object. I think the registration have to happen at another place.

Reply:
It seems module constructors can be used for setup: https://www.bennadel.com/blog/3180-ngmodule-constructors-provide-a-module-level-run-block-in-angular-2-1-1.htm

I also checked and confirmed now that the constructor is only called once even if the module is imported multiple times.